### PR TITLE
[Dynamo][autograd.Function] Should graph break on stride accesses in backward

### LINKED ISSUE #118399: Dynamo x autograd.Function: stride accesses in backward should cause graph break
```
import torch

foo = []

class MulY(torch.autograd.Function):
    @staticmethod
    def forward(ctx, x):
        ctx.set_materialize_grads(False)
        return x * 3

    @staticmethod
    def backward(ctx, grad_out):
        if grad_out.stride(0) == 1:
            foo.append(grad_out)
            return grad_out * 2
        return grad_out * 3

@torch.compile(backend="eager")
def f(x):
    return MulY.apply(x)

x = torch.tensor(2., requires_grad=True)
expected = MulY.apply(x)
out = f(x)
assert torch.allclose(out, expected)
```

It looks like we lost this at some point, because now it's just a loud error.

cc @ezyang @msaroufim @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #118399

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/dynamo/test_autograd_function.py [hunk 1]
```diff
@@ -281,7 +281,7 @@ def test_stride_in_bwd(self):
         x = torch.randn(2, 2, dtype=torch.double, requires_grad=True)
         with self.assertRaisesRegex(
             torch._dynamo.exc.Unsupported,
-            ".*HigherOrderOperator body's output must consist of tensors only",
+            "Illegal getattr invocation stride in strict mod",
```
Comment (yanboliang) [inline test/dynamo/test_autograd_function.py:284]: This was accidentally changed by https://github.com/pytorch/pytorch/commit/28bb31e4a588f3fa198e927439fe09c871201aab#diff-6604f3e9a4e51c5e65634ca48b897932ac03b1837efb27a3f3b3948f99567c6bL278

File: torch/_dynamo/variables/misc.py [hunk 2]
```diff
@@ -410,38 +406,6 @@ def call_method(
...
-                    self.fn_cls.forward.__func__,
-                    variables.UserDefinedClassVariable(self.fn_cls),
-                )
-                args = [forward.obj] + args
-            else:
-                unimplemented(
-                    f"forward is a non-function or method: {self.fn_cls.forward}"
-                )
-
-            return tx.inline_call(tx, forward, args, kwargs)
```
Comment (yanboliang) [inline torch/_dynamo/variables/misc.py:444]: I think these are dead code after #116897, as the ```forward/backward``` functions tracing were triggered in ```AutogradFunctionApplyVariable.call_function```.

File: torch/_dynamo/variables/misc.py [hunk 3]
```diff
@@ -410,38 +406,6 @@ def call_method(
                 )
             else:
                 return self.call_apply(tx, args, kwargs)
-        elif name == "backward":
-            with tx.strict_translation_mode():
```
Comment (oulgen) [inline torch/_dynamo/variables/misc.py:414]: ah nice find, i did not notice this code before

**Review by oulgen — APPROVED** [review]

File: test/dynamo/test_autograd_function.py (hunk 1 above)
Comment (zou3519) [reply test/dynamo/test_autograd_function.py:284]: Is it possible to add a stricter test so we don't regress on this in the future?

Comment (yanboliang) [reply test/dynamo/test_autograd_function.py:284]: Do you mean a stricter test that not relying on exception message? I think we can. Let me add it in a follow up PR.

Comment (zou3519) [reply test/dynamo/test_autograd_function.py:284]: Yes

Comment (yanboliang) [reply test/dynamo/test_autograd_function.py:284]: https://github.com/pytorch/pytorch/pull/119237

**Pull request closed without merging**